### ISSUE TITLE
Improve UnusedPrivateMember rule

### ISSUE THREAD

**Issue by igorwojda** [arturbosch/detekt#1764]
Works fine:
```
private fun onStateChanged(state: PaymentErrorViewState) {

}
```

Violation not detected:
```
private fun onStateChanged(state: PaymentErrorViewState) {
    state
}
```

This is a tricky case, but since we do nothing with a member we can treat it as unused as well

**Comment by schalkms** [comment]
Thank you for your feedback!
`UnusedPrivateMember` should comply with IntelliJ’s rule.
Technically, the parameter in your example is used.
There are many more examples like yours. Introducing this would unnecessarily add complexity to this rule.
Due to the mentioned reasons I close this issue.
